app/routes/search.tsx

```diff
@@ -17,11 +17,25 @@ export const loader: LoaderFunction = async ({ request }) => {
     });
   }
 
-  const events = await db.event.findMany({ include: { group: true } });
+  // for pagination in the future, use the "skip" property in conjunction with the "take" property
+  const events = await db.event.findMany({
+    include: { group: true },
+    where: {
+      OR: [
+        { location: { contains: query } },
+        { name: { contains: query } },
+        { description: { contains: query } },
+        {
+          group: {
+            OR: [{ description: { contains: query } }, { name: { contains: query } }],
+          },
+        },
+      ],
+    },
+    take: 24,
+  });
 
-  const filteredEvents = events.filter((e) => e.description.toLowerCase().includes(query.toLowerCase()));
-
-  return { events: filteredEvents, query };
+  return { events: events, query };
```

Comment:
Just for consistency:

```suggestion
  return { events, query };
```